proposals/semi-auto-properties.md

```diff
@@ -156,6 +156,64 @@ public class Point
     - https://github.com/dotnet/csharplang/issues/5563
     - https://github.com/dotnet/csharplang/pull/5573#issuecomment-1002110830
 
+### Feature name
+
+Some options for the name of the feature:
+1. semi-auto properties
+1. field-backed properties
+1. field keyword
+
+### `field` in property initializer
+
+Should `field` be a keyword in a property initializer and bind to the backing field?
+
+```csharp
+class MyClass
+{
+    private const int field = -1;
+
+    public object Property { get; } = field; // bind to const (ok) or backing field (error)?
+}
+```
+
+In the example above, binding to the backing field should result in an error: "initializer cannot reference non-static field".
```

Comment:
Is there an example where field binding to the backing property would have desirable behavior?

Reply:
I haven't found an example where binding to the backing field could be used in the property initializer.

Reply:
If so then why do we recommend making it a keyword? Seems like a breaking change that isn't balanced with a positive scenario.

Reply:
Also, if we did make `field` a keyword in property initializer, I would expect to actually be able to read the `field` in the initializer if the property is static. I feel like it gets wonky.

Reply:
Got a weirder one, which isn't on the list (and I don't know that I've seen it mentioned elsewhere).  What about properties with expression body definitions?

```cs
public int X => Increment(ref field);

private int Increment(ref int field) => field++;
```

Reply:
Yeah those are property accessors and we expect field to be a keyword in them.

Reply:
It seems like the backing field of a `static` property might be usable in the property initializer.
```csharp
class MyClass
{
    static object Property { get; } = Initialize(ref field);
    
    static object Initialize(ref object obj)
    {
        return obj = new Random().Next();
    }
}
```

Reply:
Yeah that's wonky and not useful IMO (you're already assigning the field, why do you need to be able to otherwise use it?). So put me in for wanting 'field' to just not be a keyword in the initializer, even though it might feel a little confusing when an accessor is using 'field' keyword in the same property.

Reply:
> Got a weirder one, which isn't on the list (and I don't know that I've seen it mentioned elsewhere). What about properties with expression body definitions?

@HaloFour This was part of the intended design from the beginning, supporting lazy initialization such as that example and `=> field ??=`. https://github.com/dotnet/csharplang/pull/8069 fleshes out the proposal with examples like these.